### ISSUE TITLE
TF 2.7.0: Failed to serialize tf.keras.layers.RandomFlip

### ISSUE THREAD

**Issue by sebastianlutter** [keras-team/keras#16085]
**System information**.
- Ubuntu 20.04
- TensorFlow installed from binary
- TensorFlow version 2.7.0
- Python version: 3.8.10
- GPU model and memory: 2 x Geforce RTX 3090, 24GB

**Describe the problem**.

I'm not able to serialize a model that contains `tf.keras.layers.RandomFlip`

**Describe the current behavior**.

Serialization fails with:
```
Traceback (most recent call last):
  File "minimal_example.py", line 14, in <module>
    model.save("/tmp/foo")
  File "/home/pixolution/.local/lib/python3.8/site-packages/keras/utils/traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/home/pixolution/.local/lib/python3.8/site-packages/tensorflow/python/saved_model/function_serialization.py", line 67, in serialize_concrete_function
    raise KeyError(
KeyError: "Failed to add concrete function 'b'__inference_sequential_layer_call_fn_257'' to object-based SavedModel as it captures tensor <tf.Tensor: shape=(), dtype=resource, value=<Resource Tensor>> which is unsupported or not reachable from root. One reason could be that a stateful object or a variable that the function depends on is not assigned to an attribute of the serialized trackable object (see SaveTest.test_captures_unreachable_variable)."
```

**Describe the expected behavior**.

Model is properly serialized

**Standalone code to reproduce the issue**.

```
import tensorflow as tf

model = tf.keras.Sequential([
    tf.keras.layers.Input(shape=(96, 96, 3)),
    tf.keras.layers.RandomFlip("horizontal")
])

model.compile(
    optimizer=tf.keras.optimizers.Adam(),
    loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    metrics=[tf.keras.metrics.SparseCategoricalAccuracy(name="acc")],
)

model.save("/tmp/foo")
```

**Source code / logs**.

After some trial and error it seems that this line in `keras/layers/preprocessing/image_preprocessing.py` is causing the issue:
```
seed = self._random_generator.make_seed_for_stateless_op()
```

* https://github.com/keras-team/keras/blob/2c48a3b38b6b6139be2da501982fd2f61d7d48fe/keras/layers/preprocessing/image_preprocessing.py#L387

When I replace this line with `seed = (42, 1337)` the model can be saved without issues. The `self._random_generator.make_seed_for_stateless_op()` function creates the `<tf.Tensor: shape=(), dtype=resource, value=<Resource Tensor>>` object that is unsupported or cannot be reached. However, to fix the seed is not a good solution.

For now I use a copy of `RandomFlip` class that uses a fixed seed, but how can I solve this in a better way? Thanks for any help

**Comment by jvishnuvardhan** [comment]
@sebastianlutter Looks like this was already resolved in `TF2.8`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/aa7d87f5c694083bdaa4615840ddc4ac/untitled.ipynb) is a gist for reference.

Can you please verify once and close the issue if this was resolved for you? Thanks

**Comment by sebastianlutter** [comment]
The problem is fixed in tensorflow 2.8.0
